rest_blog/views.py

```diff
@@ -0,0 +1,52 @@
+from django.views.decorators.csrf import csrf_exempt
+from django.http import HttpResponse, JsonResponse
+from rest_framework.parsers import JSONParser
+from rest_framework.decorators import api_view
+from blog.models import Reminder
+from rest_blog.serializers import ReminderSerializer
+
+
+@csrf_exempt
+def reminder_list(request):
+    """
+    List all Reminders, or create a new Reminder.
+    """
+    if request.method == 'GET':
+        reminders = Reminder.objects.all()
+        serializer = ReminderSerializer(reminders, many=True)
+        return JsonResponse(serializer.data, safe=False)
+
+    elif request.method == 'POST':
+        data = JSONParser().parse(request)
+        serializer = ReminderSerializer(data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return JsonResponse(serializer.data, status=201)
+        return JsonResponse(serializer.errors, status=400)
+
+
+@csrf_exempt
+def reminder_detail(request, pk):
+    """
+    Retrieve, update or delete a Reminder.
+    """
+    try:
+        reminder = Reminder.objects.get(pk=pk)
+    except Reminder.DoesNotExist:
+        return HttpResponse(status=404)
+
+    if request.method == 'GET':
+        serializer = ReminderSerializer(reminder)
+        return JsonResponse(serializer.data)
+
+    elif request.method == 'PUT':
+        data = JSONParser().parse(request)
+        serializer = ReminderSerializer(reminder, data=data)
```

Comment:
@vamsikrishna-kolli You don't have to use `JSONParser `. Just pass `request.data` to the `ReminderSerializer`.